RA-05-create-fragment-for-favorites

DenisGubernatorov opened the pull request:
1) Добавлен класс для категории Избранное, реализующий Fragment. 2) Добавлены лэйаут fragment_favorites.xml для категории Избранное 3) В MainActivity настроены слушатели кнопок. При нажатии в контейнере FragmentContainerView размещается соответствующий фрагмент

DenisGubernatorov committed:
RA-05-create-fragment-for-favorites
1) Добавлен класс для категории Избранное, реализующий Fragment.
2) Добавлены лэйаут fragment_favorites.xml для категории Избранное
3) В MainActivity настроены слушатели кнопок. При нажатии в контейнере FragmentContainerView размещается соответствующий фрагмент

ievetrovdev commented on app/src/main/res/values/strings.xml:
```diff
@@ -3,4 +3,5 @@
     <string name="text_category">Категории</string>
     <string name="text_favorites">Избранное</string>
     <string name="text_main_fragment">Основной фрагмент</string>
+    <string name="text_favorites_fragment">Избранное</string>
```
Повтор ресурса

ievetrovdev commented on app/src/main/java/com/example/recipesapp/MainActivity.kt:
```diff
@@ -14,10 +15,27 @@ class MainActivity : AppCompatActivity() {
         supportFragmentManager.commit {
             setReorderingAllowed(true)
-            add(R.id.mainContainer, categoriesListFragment)
+            replace(R.id.mainContainer, fragment)
```
Ты используешь Java метод add/replace из класса FragmentTransaction.java (посмотри его декларацию). Документация же предлагает импользовать Kotlin extension функцию add() из класса FragmentTransaction.kt. 

Именно для импорта метода add/replace нужно было имплементировать библиотеку fragment-ktx. Экстеншн подразумевает принудительное указание типа, а второй параметр с созданием экземпляра фрагмента уже не нужен.

DenisGubernatorov committed:
RA-05-create-fragment-for-favorites
1) Удаление лишнего строкового ресурса
2) Замена использования java метода replace из FragmentTransaction.java на функцию-расширение.

aGroupoid commented on app/src/main/java/com/example/recipesapp/MainActivity.kt:
```diff
@@ -17,24 +18,22 @@ class MainActivity : AppCompatActivity() {
 
         if (savedInstanceState == null) {
-            loadFragment(CategoriesListFragment())
+            load<CategoriesListFragment>()
```
Сейчас стартовый экран тоже получает свойство addToBackStack(null). Такого нам не надо, поскольку при нажатии назад на стартовом экране, пользователь  попадёт в непонятый стейт с пустым экраном без фрагментов. Давай расширим функцию load булевым атрибутом, который по умолчанию true. И добавим условие, добавлять в бэкстек только когда true. В этом месте закидывать в функцию значение false

aGroupoid reviewed (approved):
Основная логика написано верно, молодец. Добавь фикс при выполнении следующего ПР
